Sources/Segment/Utilities/Networking/HTTPSession+Apple.swift

```diff
@@ -0,0 +1,22 @@
+import Foundation
+
+#if os(Linux) || os(Windows)
+import FoundationNetworking
+#endif
+
+extension URLSessionDataTask: DataTask {}
+
+// Give the built in `URLSession` conformance to HTTPSession so that it can easily be used
+extension URLSession: HTTPSession {
+  public func uploadTask(with request: URLRequest, fromFile file: URL, completionHandler: @escaping (Data?, URLResponse?,  Error?) -> Void) -> any DataTask {
+      let task: URLSessionUploadTask = uploadTask(with: request, fromFile: file, completionHandler: completionHandler)
+
+      return task
+  }
+
+  public func dataTask(with request: URLRequest, completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void) -> any DataTask {
+    let task: URLSessionDataTask = dataTask(with: request, completionHandler: completionHandler)
+
+    return task
+  }
```

Comment:
The compiler seemed to not find the right functions automatically and I needed to explicitly add the typing so it knew which function I was calling into. Maybe something in the signatures is incorrect?